src/Form/ConstraintForm.php

```diff
@@ -0,0 +1,233 @@
+        }
+
+        if (array_key_exists('descriptionPlural', $input)) {
+          unset($input['descriptionPlural']);
+        }
+
+        $form['wrapper']['descriptionSingular']['#default_value'] = $plugin_instance->defaultDescriptionSingular();
+        $form['wrapper']['descriptionPlural']['#default_value'] = $plugin_instance->defaultDescriptionPlural();
+      }
+      else {
+        $form['wrapper']['descriptionSingular']['#default_value'] = $this->entity->getDescriptionSingular();
+        $form['wrapper']['descriptionPlural']['#default_value'] = $this->entity->getDescriptionPlural();
+      }
+
+      // Get plugin specific form elements if type is set.
+      $form['wrapper']['settings'] += $plugin_instance->settingsForm();
+    }
+    else {
+      // Show an empty message if there is no type selected yet.
+      $form['wrapper']['settings']['empty'] = [
+        '#type' => 'item',
+        '#markup' => $this->t('Please select a constraint.'),
+      ];
+    }
+
+    return $form['wrapper'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    $type = $form_state->getValue('type');
+
+    if ($this->entity->isNew()) {
+      // If the constraint marked as unique, make sure that only one can be
+      // created for a given policy.
+      $plugin_definition = $this->passwordConstraintPluginManager->getDefinition($type);
+      if (!empty($plugin_definition['unique'])) {
+        $entities = $this->constraintConfigEntityStorage->loadByProperties([
+          'type' => $type,
+        ]);
+        if (array_key_exists($type . '.' . $this->getRequest()->get('password_enhancements_policy')->id(), $entities)) {
+          $form_state->setError($form['type'], $this->t('Only one instance can be created from the selected constraint.'));
+        }
+      }
+    }
+
+    // Validate plugin specific form elements.
+    if (!array_key_exists('empty', $form['wrapper']['settings'])) {
+      $plugin_instance = $this->passwordConstraintPluginManager->createInstance($type, $this->entity->getConfiguration());
+      $plugin_instance->settingsValidate($form['wrapper']['settings'], $form_state);
+    }
+
+    parent::validateForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    // Call plugin's submit callback.
+    if (!array_key_exists('empty', $form['wrapper']['settings'])) {
+      $plugin_instance = $this->passwordConstraintPluginManager->createInstance($form_state->getValue('type'), $this->entity->getConfiguration());
+      $plugin_instance->settingsSubmit($form['wrapper']['settings'], $form_state);
+    }
+
+    parent::submitForm($form, $form_state);
+
+    $this->messenger->addStatus('The password constraint was successfully saved.');
+    $form_state->setRedirect('entity.password_enhancements_constraint.collection', [
```

Comment:
`$this->getRequest()->get('password_enhancements_policy')->toUrl('collection')`

Reply:
Does not work, because the constraint does not have an ID yet when creating a new one, and I would have to redirect the user back to the constraint collection page.

Reply:
Because you should have override `save()` instead of `submitHandler` as per: `\Drupal\Core\Entity\EntityFormInterface::save()` (doc block).